src/ast/ast.c

```diff
@@ -73,7 +74,7 @@ static void _AST_LimitResults(AST *ast, const cypher_astnode_t *root_clause,
 }
 
 static void _extract_params(const cypher_astnode_t *statement) {
-	rax *params = raxNew();
+	rax *params = NULL;
 	uint noptions =  cypher_ast_statement_noptions(statement);
```

Comment:
How about we have `if (noptions == 0) return` here, then initialize `params` and delete the conditional logic elsewhere in this function?